### ISSUE TITLE
undo is not working in text objects

### ISSUE THREAD

**Issue by milahu** [uxbox/uxbox#1479]
### Reproduce

1\. create a text object
2\. insert some text
3\. hit Ctrl+Z (undo shortcut/hotkey)

### Actual

when i hit Ctrl+Z with cursor in the text object, nothing happens.

when i click outside the artboard (to remove focus from the text object),
Ctrl+Z works, but the edit-history is not fine-grained,
so Ctrl+Z removes ALL input i made when the text-box was focussed.

### Expected

fine-grained edit-history for text objects.

but probably not always - only when focus/cursor is inside the text box.

### Environment

https://design.penpot.app/

html source says

```js
window.penpotVersion = "1.10.3-beta-6102-gfdb1c5e1f";
```

**Comment by milahu** [comment]
duplicate of #1025
should we keep this open? to split 1025 into separate issues

> [D]This is an enhancement suggestion
in text editing mode, it should provide undo option for each step of adding or deleting, otherwise when editing a large text, it is impossible to undo single step operation, only all of them can be undone.

> And we found [D] to be a very reasonable suggestion, so we added to our backlog as an enhancement: https://tree.taiga.io/project/penpot/us/1830?no-milestone=1

**Comment by elhombretecla** [comment]
Hi @milahu, this is not a bug. It is a functionality to be developed in a close future :)

He is the US about it [https://tree.taiga.io/project/penpot/us/1830?backlog-q=undo&no-milestone=1](https://tree.taiga.io/project/penpot/us/1830?backlog-q=undo&no-milestone=1)

**Comment by niwinz** [comment]
> duplicate of #1025 should we keep this open? to split 1025 into separate issues

yes, we prefer small and concise issues instead one single with varios problems

**Comment by NusryNizam** [comment]
Not only in text, but as a whole, undo stack is so unpredictable, I lost my work/progress just because of this. Hope this will be resolved soon.

**Comment by Simon-Laux** [comment]
> Hi @ milahu, this is not a bug. It is a functionality to be developed in a close future :)

It is very annoying to the user experience so I would call it a bug? I guess technically it might not be implemented at all so could be then considered an enhancement? But even then it degrades UX so I personally would still call it a bug.
Is there some definition of what the term "bug" means in this project?

Also I just experienced this "missing feature", I love penpot, but the strange/unpredictable/'unstable' undo-stack is really frustrating, losing work is never fun :cry:. Though overall I'm looking positively into the future, penpot has huge potential and is already very useful.
